Source/Core/VideoBackends/Software/TransformUnit.cpp

```diff
@@ -253,81 +248,58 @@ static void LightColor(const Vec3 &pos, const Vec3 &normal, u8 lightNum, const L
 			float cosAtt = light->cosatt.x + (light->cosatt.y * attn) + (light->cosatt.z * attn * attn);
 			float distAtt = light->distatt.x + (light->distatt.y * dist) + (light->distatt.z * dist2);
 			attn = SafeDivide(std::max(0.0f, cosAtt), distAtt);
+			break;
 		}
-		else if (chan.attnfunc == 1) // specular
-		{
-			// donko - what is going on here?  655.36 is a guess but seems about right.
-			attn = (light->pos * normal) > -655.36 ? std::max(0.0f, (light->dir * normal)) : 0;
-			ldir.set(1.0f, attn, attn * attn);
-
-			float cosAtt = std::max(0.0f, light->cosatt * ldir);
-			float distAtt = light->distatt * ldir;
-			attn = SafeDivide(std::max(0.0f, cosAtt), distAtt);
-		}
-		else
-		{
+		default:
 			PanicAlert("LightColor");
-			return;
-		}
-
-		switch (chan.diffusefunc)
-		{
-			case LIGHTDIF_NONE:
-				AddScaledIntegerColor(light->color, attn, lightCol);
-				break;
-			case LIGHTDIF_SIGN:
-				{
-					float difAttn = ldir * normal;
-					AddScaledIntegerColor(light->color, attn * difAttn, lightCol);
-				}
-				break;
-
-			case LIGHTDIF_CLAMP:
-				{
-					float difAttn = std::max(0.0f, ldir * normal);
-					AddScaledIntegerColor(light->color, attn * difAttn, lightCol);
-				}
-				break;
-			default: _assert_(0);
-		}
 	}
```

Comment:
Both copies of the computation of `attn` are identical; please move them into a helper function.